consumer.go

```diff
@@ -258,19 +258,25 @@ func (c *consumer) abandonBrokerConsumer(brokerWorker *brokerConsumer) {
 // By default, it logs these errors to sarama.Logger; if you want to be notified directly of all errors, set
 // your config's Consumer.Return.Errors to true and read from the Errors channel, using a select statement
 // or a separate goroutine. Check out the Consumer examples to see implementations of these different approaches.
+//
+// To terminate such a for/range loop while the loop is executing, call AsyncClose. This will kick off the process of
+// consumer tear-down & return imediately. Continue to loop until the Messages channel has been drained. If you've
```

Comment:
I think this part might be a touch unclear; the call to `AsyncClose` will return immediately, but the loop could take some time to exit depending.

Might also be worth clarifying that `AsyncClose` will actually close the `messages` channel when cleanup is done, so the `range` will stop automatically.